src/gConsent/util/access.mock.ts

```diff
@@ -43,6 +44,7 @@ type RequestVcOptions = Partial<{
   resourceOwner: string | null;
   inherit: boolean;
   purpose: UrlString[];
+  custom: CustomFields[];
```

Comment:
We are adding assertions to the VC, so the custom fields must contain both the predicate, and the object. 
It will result in a VC looking like the following: 
```
{
  id: "https://some.credential",
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://schema.inrupt.com/credentials/v1.jsonld",
    "https://vc.inrupt.com/credentials/v1",
    "https://w3id.org/security/data-integrity/v1",
    "https://w3id.org/vc-revocation-list-2020/v1",
    "https://w3id.org/vc/status-list/2021/v1",
    "https://w3id.org/security/suites/ed25519-2020/v1",
  ],
  credentialSubject: {
    id: "https://some.requestor",
    hasConsent: {
      forPersonalData: [
        "https://some.resource",
      ],
      hasStatus: "https://w3id.org/GConsent#ConsentStatusRequested",
      mode: [
        "http://www.w3.org/ns/auth/acl#Read",
      ],
      isConsentForDataSubject: "https://some.pod/profile#you",
      "https://example.org/ns/customString": "customValue",
      "https://example.org/ns/customBoolean": true,
      "https://example.org/ns/customInt": 1,
      "https://example.org/ns/customFloat": 1.1,
    },
    inbox: "https://some.inbox",
  },
  issuanceDate: "2022-02-22T00:00:00.000Z",
  expirationDate: "2022-02-23T00:00:00.000Z",
  issuer: "https://some.issuer",
  proof: {
    created: "2022-06-08T15:28:51.810Z",
    proofPurpose: "https://example.org/some/proof/purpose",
    proofValue: "some proof",
    type: "Ed25519Signature2020",
    verificationMethod: "https://example.org/some/verification/method",
  },
  type: [
    "SolidAccessRequest",
    "VerifiableCredential",
  ],
}
```

We can have `purpose` be an array of strings, or `resourceOwner` be a `string`, because we implicitly know which predicate it will be applied to. That doesn't apply to the custom fields, where both predicate and object are user-supplied.